### PR TITLE
Test shard functionalties

### PR DESCRIPTION
**Pull request by v-harihar**

### CODE REVIEW

**Review by kshithijiyer — CHANGES_REQUESTED** [review]
I see alot of pylint issues in your code that's why it seems to be failing on the build system, could you please fix them?

**Comment by v-harihar** [comment]
@kshithijiyer sure will check. Meanwhile, Can you please provide your review comments? I will address both in the next patch

File: tests/functional/shard/test_shard_functionalties.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+        # verify default shard lru limit set to 16384                          
+        self.assertEqual(vol_option_dict['features.shard-lru-limit'], '16384', "Failed"               
+                         " to validate "                                           
+                         "volume option - features.shard-lru-limit: 16384")
+
+        # verify default shard deletion rate set to 100                         
+        self.assertEqual(vol_option_dict['features.shard-deletion-rate'], '100', "Failed"               
+                         " to validate "                                           
+                         "volume option - features.shard-deletion-rate: 100")
```
Comment (kshithijiyer) [inline tests/functional/shard/test_shard_functionalties.py:89]: You can try using a loop here instead:
```   
       # verify sharding xlator is enabled
        for val, exp_val, err_msg in (('features.shard', 'on', 'features.shard: on'),
                                                       ('features.shard-block-size', '64MB', "features.shard-block-size: 64Mb")):
            self.assertEqual(vol_option_dict[val], exp_val, "Failed"         
                                        " to validate volume option - {}".format(err_msg)) 
```

File: tests/functional/shard/test_shard_functionalties.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        self.assertEqual(vol_option_dict['features.shard-block-size'], '4MB', "Failed"           
+                         " to validate "                                           
+                         "volume option - features.shard-block-size: 4MB")                                         
+        # verify shard lru limit set to 25                                       
+        self.assertEqual(vol_option_dict['features.shard-lru-limit'], '25', "Failed"               
+                         " to validate "                                           
+                         "volume option - features.shard-lru-limit: 25")
+
+        # verify shard deletion rate set to 200                         
+        self.assertEqual(vol_option_dict['features.shard-deletion-rate'], '200', "Failed"
```
Comment (kshithijiyer) [inline tests/functional/shard/test_shard_functionalties.py:109]: Same as above

File: tests/functional/shard/test_shard_functionalties.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+        self.assertEqual(vol_option_dict['features.shard-block-size'], '64MB', "Failed"    
+                         " to validate "                                        
+                         "volume option - features.shard-block-size: 64MB")          
+        # verify shard lru limit set to 25                                       
+        self.assertEqual(vol_option_dict['features.shard-lru-limit'], '16384', "Failed"    
+                         " to validate "                                        
+                         "volume option - features.shard-lru-limit: 16384")        
+                                                                                
+        # verify shard deletion rate set to 200                                 
+        self.assertEqual(vol_option_dict['features.shard-deletion-rate'], '100', "Failed"
```
Comment (kshithijiyer) [inline tests/functional/shard/test_shard_functionalties.py:131]: Same as above

**Review by kshithijiyer — CHANGES_REQUESTED** [review]
I have a few comments to reduce the number of lines. Have s look at them and let me know what you feel about them.

**Comment by balakreddy** [comment]
@v-harihar There are a lot of unused imports in the file, you can find the pylint job here: https://build.gluster.org/job/gh_glusto-tests-lint/57/console. Please resend the patch with changes.

Thanks

**Pull request closed without merging**